lib/Cake/Test/Case/Model/Datasource/DboSourceTest.php

```diff
@@ -1047,6 +1047,35 @@ public function testNotNullOnEnum() {
 		), $enumResult);
 	}
 
+/**
+ * Test for MySQL enum datatype for a list of Integer stored as String
+ *
+ * @return void
+ */
+	public function testIntValueAsStringOnEnum() {
+		if (!$this->db instanceof Mysql) {
+			$this->markTestSkipped('This test can only run on MySQL');
+		}
+		$name = $this->db->fullTableName('enum_faya_tests');
+		$query = "CREATE TABLE {$name} (faya enum('10','20','30','40') NOT NULL);";
+		$result = $this->db->query($query);
+		$this->assertTrue($result);
+
+		$EnumFayaTest = ClassRegistry::init('EnumFayaTest');
+		$enumResult = $EnumFayaTest->save(array('faya' => '10'));
+
+		$query = "DROP TABLE {$name};";
+		$result = $this->db->query($query);
+		$this->assertTrue($result);
+
+		$this->assertEquals(array(
```

Comment:
Move this assert immediately after where `$enumResult` is retrieved.

Reply:
@markstory 

[All the test](https://travis-ci.org/cakephp/cakephp/builds/458401437) passes except [This](https://travis-ci.org/cakephp/cakephp/jobs/458401456), this is in _Allowed Failures_ tests